executor/distsql.go

```diff
@@ -824,14 +827,25 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
 		sort.Sort(task)
 	}
 
-	if w.isCheckOp && handleCnt != len(task.rows) {
-		obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
-		for _, row := range task.rows {
-			handle := row.GetInt64(w.handleIdx)
-			obtainedHandlesMap[handle] = struct{}{}
+	if handleCnt != len(task.rows) {
+		if w.isCheckOp {
+			obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
+			for _, row := range task.rows {
+				handle := row.GetInt64(w.handleIdx)
+				obtainedHandlesMap[handle] = struct{}{}
+			}
+			return errors.Errorf("inconsistent index %s handle count %d isn't equal to value count %d, missing handles %v in a batch",
+				w.idxLookup.index.Name.O, handleCnt, len(task.rows), GetLackHandles(task.handles, obtainedHandlesMap))
+		}
+
+		if len(w.idxLookup.tblPlans) == 1 {
```

Comment:
Do we need to check it?

Reply:
Only when there is a single `TableScan` operator in cop task, we can expect `len(rows) == len(handles)` otherwise some rows may be abandoned by some operators like `Selection` so `len(rows) < len(handles)`.

Reply:
`len(w.idxLookup.tblPlans) == 1` should be semantically equal to `ok` and `NoCondition` in this context?

Reply:
Yes you are right.
`NoCondition` and `ok` have been removed.